### PR TITLE
Document the Clearance::Engine class

### PR DESCRIPTION
**Pull request by derekprior**

### CODE REVIEW

**Comment by BlakeWilliams** [comment]
LGTM

File: lib/clearance/engine.rb [hunk 1]
```diff
@@ -2,6 +2,21 @@
 require "rails"
 
 module Clearance
+  # Makes Clearance behavior available to Rails apps on initialization. By using
+  # a Rails Engine rather than a Railtie, Clearance can automatically expose its
```
Comment (JoelQ) [inline lib/clearance/engine.rb]: What does "automatically expose its routes" mean in this context? That we don't have to manually load them in the routes file?

Comment (derekprior) [inline lib/clearance/engine.rb]: yes.

File: lib/clearance/engine.rb [hunk 2]
```diff
@@ -2,6 +2,22 @@
...
+  # own routes and views to the hosting application.
+  #
+  # Requiring `clearance` (likely by having it in your `Gemfile`) will
+  # automatically require the engine. You can opt-out of Clearance's internal
+  # routes by using {Configuration#routes=}. You can override the Clearance
```
Comment (gylaz) [inline lib/clearance/engine.rb]: What should `Configuration#routes=` be set to in the case of opting-out? `false`, `[]`, something else?

Comment (derekprior) [inline lib/clearance/engine.rb]: `false`. In the rendered docs, this is a link to the documentation for that method, which says:

```ruby
    # Set to `false` to disable Clearance's built-in routes.
    # Defaults to `true`. When set to false, your app is responsible for all
    # routes. You can dump a copy of Clearance's default routes with
    # `rails generate clearance:routes`.
    # @return [Boolean]
    attr_writer :routes
```

**Pull request merged**